src/tree/mod.rs

```diff
@@ -101,14 +101,11 @@ where
     ///
     /// This includes padding the value to the correct branch length equivalent
     pub fn root(&self) -> io::Result<BlsScalar> {
-        let first_level: A = self.inner().annotation().ok_or_else(|| {
-            io::Error::new(
-                io::ErrorKind::Other,
-                "Kelvin shouldn't fail at reading annotations",
-            )
-        })?;
-        let storage_scalar: &StorageScalar = first_level.borrow();
-        Ok(storage_scalar.to_owned().into())
+        let branch = self
+            .poseidon_branch(0)?
+            .unwrap_or(PoseidonBranch::mock(&[], self.branch_depth as usize));
```

Comment:
I don't think this is the expected behaviour of the function. 

If `self.poseidon_branch` failed, there's no reason at all to return something that is completely mocked. Since it simply does not represent anything related to the real tree/branch.

Reply:
That unwrap is not for failing, but for being non-existent. In this case, its a `None`, not an `Err`. If the tree don't contain any elements, it will be `None`